README.md

```diff
@@ -127,7 +127,7 @@ const client = getClient();
 
 ## Ember Setup
 
-To correctly setup the Apollo client, you should use a instance initializer.
+To correctly setup the Apollo client, you should use an instance initializer.
 This will ensure you can access services, for example, to add access tokens to
 your requests or something else.
```

Comment:
Quoting Robert Jackson from https://github.com/emberjs/rfcs/issues/508#issue-462043002:

> [...] These types of implicit injects are nearly always done from initializers or instance-initializers *(both of which should also die in a 🔥, but that is another issue)*.

The general sentiment seems to be, that initializes should eventually be deprecated. They cause multiple issues, most prominently making testing harder.

What do you think about recommending setting this up in the `application` `Route` `beforeModel()` / `activate()` / `constructor()` instead?

Reply:
This is interesting. I guess I would need to figure out what is the best option. At first glance, I don't like the idea of using routes for setting up the apollo client. I do think it belongs to the application, maybe app.js ? But the issue is that I need the instance of the app so I can associate a destroyer.

Reply:
So, I have been thinking about this more, and one issue that I could see with the option to set in the application route is:
 Components using apollo would not be able to get the client because it would not be set up when running component integration tests as the application route would not run in that scenario.